### PR TITLE
fix: Test latest_dataset is not none when bounding_box is None

### LINKED ISSUE #440: Bounding box unit test issue from #431
### Describe the bug

In standup, we discussed that the issue with #431 wasn't caught because of issues with the bounding box unit test.

### Steps/Code to Reproduce

Discuss/resolve

### Expected Results

TBD

### Actual Results

The latest dataset not returning on feeds with a null bounding box was not caught by the unit tests.

### Screenshots

_No response_

### Files used

_No response_

### Additional notes

_No response_

### PR DESCRIPTION
**Pull request by jcpitre**
**Summary:**

Closes #440 
Tested the conditions that lead to #431 

**Expected behavior:** 

These tests pass: 

- https://github.com/MobilityData/mobility-feed-api/blob/8eee5c90fad8c275bc0e7a947d565c847b189d0c/api/tests/unittest/test_feeds.py#L114
- https://github.com/MobilityData/mobility-feed-api/blob/8ebbffeebcccc994deb347de40dff83968c3506b/api/tests/unittest/test_feeds.py#L199

**Testing tips:**

To test I put back the code from [PR #433](https://github.com/MobilityData/mobility-feed-api/pull/433/files) in api/src/feeds/impl/feeds_api_impl.py and verified that it failed.

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Run the unit tests with `./scripts/api-tests.sh` to make sure you didn't break anything
- [ ] Add or update any needed documentation to the repo
- [x] Format the title like "feat: [new feature short description]". Title must follow the Conventional Commit Specification(https://www.conventionalcommits.org/en/v1.0.0/).
- [x] Linked all relevant issues
- [ ] Include screenshot(s) showing how this pull request works and fixes the issue(s)

### CODE REVIEW

File: api/tests/unittest/test_feeds.py [hunk 1]
```diff
@@ -125,20 +125,7 @@ def test_gtfs_feeds_get(client: TestClient, mocker):
         subdivision_name="test_subdivision_name",
         municipality="test_municipality",
     )
-    mock_bounding_box = json.dumps(
```
Comment (davidgamez) [inline api/tests/unittest/test_feeds.py:128]: [suggestion]: to have better coverage, I suggest creating a test for the missing bounding box, leaving this test as it covers the bounding box's presence and structure.

Comment (jcpitre) [reply api/tests/unittest/test_feeds.py:128]: Actually AFAIK the bounding box is not checked in this test. 
I could add an assert for the bounding box in this test and create a new one.

Comment (davidgamez) [reply api/tests/unittest/test_feeds.py:128]: Yes please

**Review by davidgamez — APPROVED** [review]
LGTM!

**Review by davidgamez — APPROVED** [review]
LGTM!

**Pull request merged**